### PR TITLE
Update add-on detail API regexp for guid detection

### LINKED ISSUE #2786: Update add-on detail API regexp for guid detection
The add-on detail API currently uses the `r'^(\{.*\}|.*@.*)$'` regexp to determine if the identifier in the URL is a guid or a slug. It works, but the actual regexp we should be using is `^(\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\}|[a-z0-9-\._]*\@[a-z0-9-\._]+)$` (**case-insensitive**), as seen in https://dxr.mozilla.org/mozilla-central/source/toolkit/mozapps/extensions/internal/XPIProvider.jsm#275

### PR DESCRIPTION
**Pull request by diox**
This matches what Firefox uses.

Fix #2786

### CODE REVIEW

File: src/olympia/addons/views.py [hunk 1]
```diff
@@ -658,7 +658,11 @@ class AddonViewSet(RetrieveModelMixin, GenericViewSet):
               AllowReviewer, AllowReviewerUnlisted),
     ]
     serializer_class = AddonSerializer
-    addon_id_pattern = re.compile(r'^(\{.*\}|.*@.*)$')
+    addon_id_pattern = re.compile(
+        # Match {uuid} or something@host.tld ("something" being optional)
+        # guids. Copied from mozilla-central XPIProvider.jsm.
+        r'^(\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\}'
+        r'|[a-z0-9-\._]*\@[a-z0-9-\._]+)$')
```
Comment (eviljeff) [inline src/olympia/addons/views.py]: The regex definition in XPIProvider.jsm ignores case.  Should the same be done here?

File: src/olympia/addons/tests/test_views.py [hunk 2]
```diff
@@ -1559,6 +1559,11 @@ def test_get_by_guid_email_format(self):
         self.url = reverse('addon-detail', kwargs={'pk': self.addon.guid})
         self._test_detail_url()
 
+    def test_get_by_guid_email_short_format(self):
+        self.addon.update(guid='@example.tld')
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_views.py]: the common case is just `@example` - change the test or add another?

**Comment by eviljeff** [comment]
r+wc

File: src/olympia/addons/views.py (hunk 1 above)
Comment (diox) [inline src/olympia/addons/views.py]: Yes, good catch.

File: src/olympia/addons/tests/test_views.py (hunk 2 above)
Comment (diox) [inline src/olympia/addons/tests/test_views.py]: Good point, I'll add another to be sure.

**Pull request merged**